docs/specification/2025-03-26/basic/authorization.md

```diff
@@ -4,242 +4,391 @@ type: docs
+The following demonstrates the OAuth 2.1 flow for public clients using PKCE.
 
 ```mermaid
 sequenceDiagram
-    participant B as User-Agent (Browser)
-    participant C as Client
-    participant M as MCP Server
-
-    C->>M: MCP Request
-    M->>C: HTTP 401 Unauthorized
-    Note over C: Generate code_verifier and code_challenge
-    C->>B: Open browser with authorization URL + code_challenge
-    B->>M: GET /authorize
-    Note over M: User logs in and authorizes
-    M->>B: Redirect to callback URL with auth code
-    B->>C: Callback with authorization code
-    C->>M: Token Request with code + code_verifier
-    M->>C: Access Token (+ Refresh Token)
-    C->>M: MCP Request with Access Token
-    Note over C,M: Begin standard MCP message exchange
+    participant UA as User-Agent (Browser or App/Broker)
+    participant MC as MCP Client
+    participant MS as MCP Server
+    participant IDP as Identity Provider
+
+    MC->>MS: MCP Request
+    MS-->>MC: 401 Unauthorized with issuer, scope, and resource in WWW-Authenticate
+    MC->>MC: Construct OIDC /authorize endpoint from issuer identifier
+    Note over MC: Generate code_verifier and code_challenge,<br/>and request with scope and resource
+    MC-->>UA: Open /authorize
+    Note over UA: User authenticates to the IDP
+    UA->>MC: Return auth code to client
+    MC->>IDP: Exchange auth code for tokens
+    IDP-->>MC: Return tokens
+    MC->>MC: Cache tokens
+    Note over MC, MS: Begin standard MCP message exchange
 ```
 
 ### 2.3 Server Metadata Discovery
 
+Servers **MUST** return the issuer identifier in `WWW-Authenticate` headers in a 
+*HTTP 401 Unauthorized* response.
+
+For servers that are overriding identity provider capabilities, such as those for token exchange or 
+dynamic client registration, they **MUST** return their fully-qualified domain as the issuer in 
+`WWW-Authenticate` headers in a *HTTP 401 Unauthorized* response.
+
 For server capability discovery:
 
-- MCP clients _MUST_ follow the OAuth 2.0 Authorization Server Metadata protocol defined
-  in [RFC8414](https://datatracker.ietf.org/doc/html/rfc8414).
-- MCP server _SHOULD_ follow the OAuth 2.0 Authorization Server Metadata protocol.
-- MCP servers that do not support the OAuth 2.0 Authorization Server Metadata protocol,
-  _MUST_ support fallback URLs.
+- MCP clients **MUST** parse the `WWW-Authenticate` response to discover the issuer URL, and construct 
+  the metadata endpoint from that issuer identifier.  
+- MCP servers **MUST** use an established identity provider and include its issuer in 
+  *HTTP 401 Unauthorized* responses.  
+- MCP clients **SHOULD** evaluate or enable the user to evaluate the trustworthiness of identity 
+  providers.  
+- MCP servers with partial support for other identity providers or that implement their own 
+  authentication and authorization logic **SHOULD** follow the OAuth 2.0 Authorization Server Metadata 
+  protocol.  
```

Comment:
I think `partial support for other identity providers` is confusing here. What about:

```suggestion
- Identity providers **MUST** support [OAuth 2.0 Authorization Server Metadata](https://datatracker.ietf.org/doc/html/rfc8414).
  protocol.  
```